### ISSUE TITLE
Looking for additional maintainers

### ISSUE THREAD

**Issue by djhoese** [vispy/vispy#2570]
Hi everyone,

I'm one of the core maintainers of vispy. I'm looking for some help. My one project that depended on vispy and was my only reason for using it and keeping it going has been unfunded for years. Other funding (ex. CZI OSS) was acquired and helped the project out for a while but that is gone now too. Ultimately I don't have the time to keep this project moving along at the moment. I've been volunteering my time when I can to help and answer questions, but that volunteer time is going to disappear really soon as I'm expecting my second kid in early March. So if there is anyone who has some spare cycles to help maintain vispy comment here or reach out on the mailing list or email me directly if you can find my email (I honestly forget to check gitter so that might not be the best place).

I'd like to note that I am not an expert at OpenGL and I'm not even an expert at VisPy. I have to look up and test out scripts before answering most questions. You don't need to be an expert at this stuff. You mostly just need to be around when you can.

Here is what I'm hoping someone could help the existing maintainers out with in order from highest importance to lesser importance:

1. Answering github issues, stackoverflow questions, and gitter questions.
2. CI fixes and updates: CI is currently working (I think), but things pop up a lot:
   a. future Python, Cython, and numpy compatibility
   b. github runner and cibuild wheel updates
   c. random surprises
3. Reviewing pull requests
4. Improve documentation to avoid repeatedly filed issues (FAQs?)

I'm still around just...not much.

**Comment by rougier** [comment]
@djhoese Thank you so much for your work over these years. Your immense contribution has allowed to maintain vispy for the community. I cannot maintain it myself even if I'm one of the original creator. But I'm always lurking around to answer any OpenGL questions when necessary. Whoever the new maintainer, you won't be alone.

**Comment by brisvag** [comment]
@djhoese I echo @rougiers thanks for your time and efforts these years :) I plan to still be around as I work on napari, so the project won't be left alone. I also can't really commit to a "full" maintainer effort (whatever that means ^^), but I'll try to raise awareness about the need of maintainance time here and hopefully we can get more people on board!

**Comment by kevinyamauchi** [comment]
Thank you for all of the work you have done maintaining vispy., @djhoese ! Best of luck with the expanding family!

**Comment by aganders3** [comment]
Thanks @djhoese for all your work, and specifically for patiently helping me get some contributions into the project.

I am happy to help where I can. My second child is now 3, so time is not _quite_ as scarce. I will turn on some notifications for the repo and (try) to keep an eye on SO and Gitter.

**Comment by rossant** [comment]
Hi David,

A huge thank you for your relentless efforts during all these years. The project is still alive in great part due to your continued investment to maintain the project, to welcome new users, to answer questions and so much more. As one of the cofounders of VisPy, I'm really amazed to see where the project is 11 years later and it's absolutely thanks to you and all the other people involved. I very much know how hard it is to maintain open-source software so I'm all the more grateful.

I've been away from the codebase for many years and I'm not familiar with the layers that are most useful to the users, so I don't think I'll be able to be a maintainer of VisPy 1.x. However, I still have a major interest in scientific visualization, of course, which is why I've been working for several years on [Datoviz](https://datoviz.org/), and on the CZI-funded VisPy 2 project with @rougier.

VisPy 2 will have a totally different architecture, as the low-level GPU rendering details and the high-level scientific visualization logic will be totally decoupled. In fact, visualizations will be described in terms of graphical primitives and data processing without assuming anything about the underlying renderer. Multiple renderers implementing the same intermediate-level graphics protocol will be able to render the same scene using possibly wildly different rendering technologies (and in fact not necessarily GPU-based).

Decoupling the GPU parts from the scientific plotting parts was also the idea of VisPy 1, however the technology at the time didn't let us achieve this without sacrificing performance. Now, with modern graphics APIs such as Vulkan, WebGPU, etc, we do have the technology for this.

This decoupling will considerably help the project to get more contributors and maintainers, because most of the work is going to be in the higher-level layers that do not require any GPU knowledge at all. With @rougier we take care of the low-level GPU parts, we have worked on this for 10+ years (including academic publications in computer graphics journals for Nicolas), and we're taking great care ensuring this low-level rendering technology can be used seamlessly from the high-level interfaces.

We're finalizing a whitepaper describing this architecture along with the implementation in Python and code examples, and we can't wait sharing them with the community in the next couple of months.

**Comment by almarklein** [comment]
Hey David, I wish you and your family all the best. Time is precious, and it gets all the more precious as your `family_size` increases ;) Setting priorities gets all the more important.

You've done a great job leading this project over the past years. Not the easiest software to manage, more so because we left quite a few open ends ... 

I'm putting my efforts in pygfx because I think wgpu is the future (and it helps being funded to work on it). I'm interested to see what vispy2 will bring though!

**Comment by kmuehlbauer** [comment]
Dear David,

thanks for your all your work leading the project for almost 7 years now (see https://github.com/vispy/vispy/issues/1322). I've very much enjoyed the collaboration within the VisPy dev/maintainer team. All the best for you and your growing family. As already said, I'm still mainly using Python but got dragged away from VisPy over the years. I'm just around the corner, if there is something I might know or remember.

**Comment by rossant** [comment]
> I'm putting my efforts in pygfx because I think wgpu is the future (and it helps being funded to work on it). I'm interested to see what vispy2 will bring though!

I agree, WebGPU/wgpu are very promising and we're actively working on the foundations to make VisPy 2 compatible with these technologies in the future!

**Comment by alisterburt** [comment]
@djhoese thank you for all of the time spent keeping vispy going and congratulations on your second kid!

**Comment by perdigao1** [comment]
I have some opengl experience from old projects ([https://sourceforge.net/projects/lpmolecularviewer/](url))
But I have no experience with shaders though.

**Comment by melonora** [comment]
@djhoese Thanks for all your work on Vispy all these years and congratulations on your second child!

**Comment by dprjoao** [comment]
Hello all, im starting to work using Vispy to make 3D visualization. I found this post and i can volunteer to help with trying to answer questions about Vispy, i love how it is built and would like to help maintaining it. One question is, its safe assume Vispy will keep growing and alive before Vispy 2, I feel like its cooled down a bit.

**Comment by rossant** [comment]
Hi @dprjoao thank you for your interest in VisPy! @rougier and I are making good progress on VisPy 2.0 and we'll make an announcement very soon. In the meantime, you can also look at [Datoviz 0.2](https://datoviz.org/) even though it's a bit lower level than VisPy 2.0.

**Comment by rougier** [comment]
In the meantime, you can directly use datoviz actually.

**Comment by rjsdotorg** [comment]
Is this still an active question?
If I can get vispy to speed up my current applications, I could help with it semi-regularly. Not an OpenGL person, but 20+ years of Python. I also have AI subscriptions which I now am finding quite useful for code maintenance and refactoring.

**Comment by brisvag** [comment]
@rjsdotorg contributions and help with maintenance are always welcome! It's currentlly mostly me maintaining here. Feel free to open PRs if you have useful stuff, I'll be happy to review :)
